docs/src/templates/pages/documentation/js/writing-tests.hbs

```diff
@@ -0,0 +1,304 @@
+    1. [Avoid logic in your tests](#avoid-logic-in-your-tests)
+    1. [Don't test multiple concerns in the same test](#dont-test-multiple-concerns-in-the-same-test)
+    1. [Cover the general case and the edge cases](#cover-the-general-case-and-the-edge-cases)
+    1. [Test the behaviour, not the internal implementation](#test-the-behaviour-not-the-internal-implementation)
+    1. [Don't mock everything](#dont-mock-everything)
+    1. [Create new tests for every defect](#create-new-tests-for-every-defect)
+    1. [Don't write unit tests for complex user interactions](#dont-write-unit-tests-for-complex-user-interactions)
+    1. [Test simple user actions](#test-simple-user-actions)
+    1. [Review test code first](#review-test-code-first)
+    1. [Practice code katas, learn with pair programming](#practice-code-katas-learn-with-pair-programming)
+
+<a name="principles"></a><h2 class="sg-sectionHeading">General principles</h2>
+
+<h3 class="sg-sectionHeading">Unit tests</h3>
+
+**Unit tests are isolated and independent of each other**
+
+- Any given behaviour should be specified in one and only one test
+- The execution/order of execution of one test cannot affect the others
+
+**Unit tests are lightweight tests**
+
+- Repeatable
+- Fast
+- Consistent
+- Easy to write and read
+
+**Unit tests are code too**
+
+They must meet the same level of quality as the code being tested. They can be refactored as well to make them more maintainable and/or readable.
+
+
+<h3 class="sg-sectionHeading">Design principles</h3>
+
+The key to good unit testing is to write **testable code**. Applying simple design principles can help, in particular:
+
+- Use a **good naming** and **comment** your code (the "why?" not the "how"), keep in mind that comments are not a substitute for bad naming or bad design
+- **DRY**: Don't Repeat Yourself, avoid code duplication
+- **Single responsibility**: each object/function must focus on a single task
+- Keep a **single level of abstraction** in the same component (for example, do not mix business logic with lower-level technical details in the same method)
+- **Minimize dependencies** between components: encapsulate, interchange less information between components
+- **Support configurability** rather than hard-coding, this prevents from having to replicate the exact same environment when testing (e.g.: markup)
+- Avoid global mutable state
+
+
+<a name="guidelines"></a><h2 class="sg-sectionHeading">Guidelines</h2>
+
+The goal of these guidelines is to make your tests:
+
+- **Readable**
+- **Maintainable**
+- **Trustworthy**
+
+These are the 3 pillars of good unit testing.
+
+
+<h3 class="sg-sectionHeading">Name your tests properly</h3>
+
+Tests names should be concise, explicit, descriptive and in correct English. Read the output of the spec runner and verify that it is understandable! Keep in mind that someone else will read it too. Tests can be the live documentation of the code.
+
+**:(**
```

Comment:
The good and bad practice section headings should be consistent with those on the other docs files e.g. in the [code style docs](https://github.com/justeat/global-component-library/pull/43/files#diff-f2d62a747e35a8d1646153fcb592ca47R106) we are using a `// bad` comment in the code example.

I think maybe something between the two might be better such as `**Good**`, `**Bad**`, possibly?

Reply:
I've changed these to be **Good:** and **Bad:** now for all instances.